src/job_description_feature_extraction.py

```diff
@@ -0,0 +1,49 @@
+import string
+import operator
+from collections import Counter
+
+
+def get_one_hot_encoded_words(feature_to_extract):
+    """
+    get_feature_word_count(x) returns a list of tuples which consists of word:count
+    From those lists, the words appear in the list below ( word_list ) was hand picked
+    Fhey are the word which might have an impact on the salary
+    """
+    word_list = ['excellent', 'graduate', 'immediate', 'junior', 'urgent']
+    encoded_features = []
+    for item in word_list:
+        one_hot_encoded_list = []
+        for data in feature_to_extract.iterrows():
+            sentence = data[1][0].lower()  # Lower casing the sentence
+            if item in sentence:
+                one_hot_encoded_list.append(1)
+            else:
+                one_hot_encoded_list.append(0)
+        encoded_features.append(one_hot_encoded_list)
+    return encoded_features
+
+
+def get_feature_word_count(column_to_count):
+    count_dict = {}
+    for data in column_to_count.iterrows():
+        string_data = str(data)
+        data_digits_removed = ''.join(
+            [i for i in string_data if not i.isdigit()]
+        )  # Remove digits
+        puct_exclude = set(string.punctuation)
+        data_punct_removed = ''.join(ch for ch in data_digits_removed if ch not in puct_exclude)
+        # Remove punctuations
+        data_punct_removed = data_puctuations_removed.lower()  # Lowercase the string
+        data_as_list = data_punct_removed.split()  # Turn into list
+        counts_as_dict = Counter(data_as_list)  # Turn into dictionary with counts
+        for item in counts_as_dict:
+            if item in count_dict:  # If item exists in the dict
+                count_dict[item] = count_dict[item]+1
```

Comment:
1) it helps with readbility because it is more consice and straight forward: read "i want to increase var by X" vs "i want to set var to var + X"
2) not a python thing, but generally better practice, as it helps the compiler. see https://softwareengineering.stackexchange.com/questions/134118/why-are-shortcuts-like-x-y-considered-good-practice